Library/Homebrew/linkage_checker.rb

```diff
@@ -83,6 +97,12 @@ def check_undeclared_deps
       next true if Formula[name].bin.directory?
       @brewed_dylibs.keys.map { |x| x.split("/").last }.include?(name)
     end
+    missing_deps = @broken_deps.values.map do |v|
```

Comment:
Could you use `flat_map` here to avoid the `flatten` and also possible the nested `maps`?